### ISSUE TITLE
about softbound

### ISSUE THREAD

**Issue by zyy0721** [dslab-epfl/asap#2]
It is mentioned that ASAP can work for  SoftBound in the paper (http://dx.doi.org/10.1109/SP.2015.58), and there is a small piece of relevant codes in the project, but I don't know how to enable it. 

Is there any instructions on how to work for SoftBound?

**Comment by Sjlver** [comment]
Indeed, ASAP works with SoftBound. However, setting everything up correctly
will be tricky.

The code in this repository is the newest version of ASAP, which we have
upgraded for LLVM 3.7.  The SoftBound experiments were done with an earlier
version, because SoftBound (at the time of our research) was based on LLVM
3.4. That earlier version used the SoftBound code from
https://github.com/santoshn/softboundcets-34 .

If you want to use ASAP and SoftBound together, you would need to port both
to a common LLVM version. Probably the easiest target would be LLVM 3.9.
There is a version of SoftBound for that (
https://github.com/santoshn/SoftBoundCETS-3.9), and I suspect that you
could add ASAP onto this code with only minor changes. Once this is done,
ASAP should "just work" if you add `-fsoftboundcets` to the compilation
options.

Hope this gets you started! Let me know how it goes -- I'd be happy to
update the documentation for ASAP.
